xla/debug_options_flags.cc

```diff
@@ -985,6 +987,17 @@ void MakeDebugOptionsFlags(std::vector<tsl::Flag>* flag_list,
       "that falling back to the driver can have drawbacks like using more "
       "memory and/or other bugs during compilation, so we recommend setting "
       "this flag to false."));
+  flag_list->push_back(tsl::Flag(
+      "xla_gpu_unsafe_pipelined_while_loop_annotator_rotate_right_pattern",
+      bool_setter_for(
+          &DebugOptions::
+              set_xla_gpu_unsafe_pipelined_while_loop_annotator_rotate_right_pattern),
+      debug_options
+          ->xla_gpu_unsafe_pipelined_while_loop_annotator_rotate_right_pattern(),
+      "If this option is true, then the while loop with rotate right "
+      "pattern will be considered a pipelined while loop and the "
+      "operations within the pipeline bubbles may be considered no-ops. This "
+      "is an unsafe flag."));
```

Comment:
Can you mention that this changes the semantics of collective permute?

Reply:
Done.